### PR TITLE
fix: added deprecation warning for kafka header migration

### PR DESCRIPTION
**Pull request by aryamohanan**
Added deprecation warning for kafka header migration, in next major release(v4) we will remove the configuration option for kafka header and by default the header format will be 'string'. For more details about the kafka migration see [official documentation](https://www.ibm.com/docs/en/instana-observability/current?topic=references-kafka-header-migration#phase-2)
Related [PR](https://github.com/instana/nodejs/pull/1296) 
refs INSTA-809

### CODE REVIEW

File: packages/core/src/tracing/instrumentation/messaging/kafkaJs.js [hunk 1]
```diff
@@ -532,3 +535,11 @@ function removeInstanaHeadersFromMessage(message) {
     });
   }
 }
+
+// Note: This function can be removed as soon as we finish the Kafka header migration phase2.
+function logWarningForKafkaHeaderFormat() {
+  logger.warn(
+    '[Deprecation Warning] The configuration option for specifying the Kafka header format will be removed in the ' +
+      "next major release. The default header format will be 'string'."
```
Comment (kirrg001) [inline packages/core/src/tracing/instrumentation/messaging/kafkaJs.js:543]: suggestion: Add the docs link for the kafka header migration.

Comment (kirrg001) [inline packages/core/src/tracing/instrumentation/messaging/kafkaJs.js:543]:
```suggestion
      "next major release as the format will no longer be configurable and Instana tracers will only send string headers."
```

or similar.

File: packages/core/src/tracing/instrumentation/messaging/rdkafka.js [hunk 2]
```diff
@@ -44,17 +44,23 @@ exports.activate = function activate(extraConfig) {
...
 };
 
 exports.deactivate = function deactivate() {
   isActive = false;
 };
 
-// Note: This function can be removed as soon as we finish the Kafka header migration and remove the ability to
-// configure the header format (at that point, we will only be using string headers).
+// Note: This function can be removed as soon as we finish the Kafka header migration phase 2 and remove the ability to
+// configure the header format.
```
Comment (kirrg001) [inline packages/core/src/tracing/instrumentation/messaging/rdkafka.js:58]:
```suggestion
// configure the header format. Might happen in major release v4.
```

File: packages/core/src/tracing/instrumentation/messaging/rdkafka.js [hunk 3]
```diff
@@ -65,21 +71,10 @@ function logWarningForKafkaHeaderFormat(headerFormat) {
   // Trace correlation would be broken for rdkafka senders with the header format 'binary'. If that format has been
   // configured explicitly, we log a warning and ignore the config value. The rdkafka instrumentation alwas acts as if
   // format 'string' had been configured.
-  if (headerFormat === 'binary') {
+  if (headerFormat === 'binary' || headerFormat === 'both') {
```
Comment (kirrg001) [inline packages/core/src/tracing/instrumentation/messaging/rdkafka.js:74]: issue: I do not understand this change. We do not ignore "both" in v3.

File: packages/core/src/tracing/instrumentation/messaging/rdkafka.js [hunk 4]
```diff
@@ -44,17 +44,23 @@ exports.activate = function activate(extraConfig) {
     }
     extraConfigHeader = extraConfig.tracing.kafka.headerFormat;
   }
-  logWarningForKafkaHeaderFormat(extraConfigHeader || configHeader);
+  if (extraConfigHeader || configHeader) {
```
Comment (kirrg001) [inline packages/core/src/tracing/instrumentation/messaging/rdkafka.js:47]: q: Why this change?

**Review by kirrg001 — CHANGES_REQUESTED** [review]

File: packages/core/src/tracing/instrumentation/messaging/rdkafka.js (hunk 2 above)
Comment (kirrg001) [reply packages/core/src/tracing/instrumentation/messaging/rdkafka.js:58]: Background: I usually enter e.g. "v4" in the IDE search to look for any code which needs a cleanup.

File: packages/core/src/tracing/instrumentation/messaging/rdkafka.js (hunk 3 above)
Comment (aryamohanan) [reply packages/core/src/tracing/instrumentation/messaging/rdkafka.js:74]: Binary headers are not supported in rd-kafka instrumentation only "string", see previous debug log.

File: packages/core/src/tracing/instrumentation/messaging/rdkafka.js (hunk 4 above)
Comment (aryamohanan) [reply packages/core/src/tracing/instrumentation/messaging/rdkafka.js:47]: The function should only be called if a valid Kafka header is present.

File: packages/core/src/tracing/instrumentation/messaging/rdkafka.js (hunk 3 above)
Comment (kirrg001) [reply packages/core/src/tracing/instrumentation/messaging/rdkafka.js:74]: Sorry I just noticed that you just combined the code to one condition 👍

File: packages/core/src/tracing/instrumentation/messaging/rdkafka.js (hunk 4 above)
Comment (kirrg001) [reply packages/core/src/tracing/instrumentation/messaging/rdkafka.js:47]: `configHeader` should have always a value because we set a default value in the normalize fn.
Same for https://github.com/instana/nodejs/pull/1311/files#diff-d11898bc026195d9d4ad65d0a381844c1bc6daf3187a71ad6b7feaf62123dd7eR46.

**Review by kirrg001 — APPROVED** [review]
Pre-approving

**Review by abhilash-sivan — APPROVED** [review]

File: packages/core/src/tracing/instrumentation/messaging/kafkaJs.js (hunk 1 above)
Comment (kirrg001) [reply packages/core/src/tracing/instrumentation/messaging/kafkaJs.js:543]: https://www.ibm.com/docs/en/instana-observability/current?topic=nodejs-collector-configuration#kafka-trace-correlation-headers
